sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -256,8 +256,13 @@ class Dataset[T] private[sql](
   private[sql] def resolve(colName: String): NamedExpression = {
     queryExecution.analyzed.resolveQuoted(colName, sparkSession.sessionState.analyzer.resolver)
       .getOrElse {
-        throw new AnalysisException(
-          s"""Cannot resolve column name "$colName" among (${schema.fieldNames.mkString(", ")})""")
+        val fields = schema.fieldNames
+        val extraMsg = if (fields.exists(sparkSession.sessionState.analyzer.resolver(_, colName))) {
```

Comment:
Please declare `val resolver = sparkSession.sessionState.analyzer.resolver` before line 257 because this is used twice.

Reply:
Done